### ISSUE TITLE
Failed to download trivy

### ISSUE THREAD

**Issue by scottwestover** [Azure/container-scan#66]
We are trying to integrate this action with one of our current workflows in a private repo, but our workflow keeps failing with the following error:

```
Error: Failed to download trivy from https://github.com/aquasecurity/trivy/releases/download/v0.16.0/trivy_0.16.0_Linux-64bit.tar.gz
```

I went ahead and enabled additional logging by setting the `ACTIONS_RUNNER_DEBUG` and `ACTIONS_STEP_DEBUG` secrets to `true`, and we are seeing the following logs:

```
Could not find allowedlist file.
##[debug]Downloading https://api.github.com/repos/aquasecurity/trivy/releases/latest
##[debug]Downloading /home/runner/work/_temp/8472635f-cf05-4075-a5c2-b755d65851c4
##[debug]download complete
##[debug]isExplicit: 0.16.0
##[debug]explicit? true
##[debug]checking cache: /opt/hostedtoolcache/trivy/0.16.0/x64
##[debug]not found
##[debug]Could not find trivy in cache, downloading from https://github.com/aquasecurity/trivy/releases/download/v0.16.0/trivy_0.16.0_Linux-64bit.tar.gz
##[debug]Downloading https://github.com/aquasecurity/trivy/releases/download/v0.16.0/trivy_0.16.0_Linux-64bit.tar.gz
##[debug]Downloading /home/runner/work/****/****/_temp/tools/trivy
Error: Failed to download trivy from https://github.com/aquasecurity/trivy/releases/download/v0.16.0/trivy_0.16.0_Linux-64bit.tar.gz
##[debug]Node Action run completed with exit code 1
##[debug]Finishing: Scan Docker Image
```

Additional information about environment:
```
Current runner version: '2.277.1'
Operating System
  Ubuntu
  20.04.2
  LTS
Virtual Environment
  Environment: ubuntu-20.04
  Version: 20210208.0
  Included Software: https://github.com/actions/virtual-environments/blob/ubuntu20/20210208.0/images/linux/Ubuntu2004-README.md
```

By chance, has anyone else ran into this issue before?

**Comment by shigupt202** [comment]
@scottwestover Are you using a private runner to run this workflow? It may be possible that the firewall rules on the runner are not allowing the action to download the tool. If you're using GitHub hosted runners, then this shouldn't be the problem.

**Comment by ds-ms** [comment]
@shigupt202 if that's the case, we should consider adding support for downloading this behind a proxy?

**Comment by scottwestover** [comment]
@shigupt202 Thanks for the quick response. We are using the GitHub hosted runner. In this particular repo, we are trying to run the action on a branch that is not the default branch for the repo. After some more testing, I was able to get the action to work correct on the default branch for the repo, but if I have a workflow that runs on a non default branch, the action fails to run.

As an example, if I have the following configuration:
```
name: Scott Test

on:
  push:
    branches:
      - develop

jobs:
  build:
    name: Build and Test
    runs-on: ubuntu-latest
    strategy:
      matrix:
        tf_version: [ "0.14.2" ]
      fail-fast: true

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Scan Docker Image
        uses: azure/container-scan@v0
        with:
          image-name: foo:bar
```

The action fails to download `trivy`.

However if I use the following configuration:
```
name: Scott Test 2

on:
  push:
    branches:
      - master

jobs:
  build:
    name: Build and Test
    runs-on: ubuntu-latest
    strategy:
      matrix:
        tf_version: [ "0.14.2" ]
      fail-fast: true

    steps:
      - name: Checkout
        uses: actions/checkout@v2

      - name: Scan Docker Image
        uses: azure/container-scan@v0
        with:
          image-name: foo:bar
```

The action successfully downloads `trivy`, and proceeds to fail due to the fake image.

I tried recreating this issue in a separate repo, but I have not been able to reproduce it.

For the time being, I have moved the `container-scan` action to be part of our workflow in our `master` branch and I have commented it out of our workflow for our `develop` branch.

**Comment by shigupt202** [comment]
Branch shouldn't have any corelation with downloading of the tool. Can you confirm whether this issue is intermittent on the affected branch, or it happens every time? And is it working in other repos?
I'm not sure what caused the issue, but we can add some extra information in the logs about the error.

**Comment by scottwestover** [comment]
Unfortunately, the issue happens every time on this particular branch, and this action works correctly in the other repos that I have utilized it in. I've tried to recreate the issue, and have not been able to do so successfully.

I do appreciate the quick responses, and the help you have given. I think adding additional logs would be helpful. I can work on submitting a PR for this.

**Comment by shigupt202** [comment]
Thanks! All you need to do is include the error message in the logs here: https://github.com/Azure/container-scan/blob/master/src/trivyHelper.ts#L73

**Comment by shigupt202** [comment]
@scottwestover Can you please try again now, and share the logs? I have added the logic to include error message in logs.

**Comment by shigupt202** [comment]
Thanks for the new logs. It looks like when Trivy version changed, the action was not able to find it in the cache and tried to download the new version on the path where old version was already present (This issue should be happening with workflows running on self-hosted runners only). I'll create a PR to fix this.
Can you confirm whether the action was working earlier? (From what I understand, it should have been working when Trivy was still on it's older version)

**Comment by scottwestover** [comment]
Thanks for the update!

No, we are not using self hosted runners, we have everything going through the GitHub runner. We have this action setup on multiple repos, and this issue has only happened in the one repo, otherwise it has been working flawlessly without any issues.

This one repo was setup a little bit differently than our other repos since we were trying a few new CI workflows. I went through the history on the repo, and it looks like when we added the action to this one repo on the particular branch, it failed from the start with the download error. I'm still double checking the logs to see if it was ran on any other branch though.

**Comment by shigupt202** [comment]
Sorry for the misinterpretation. I was under the assumption that you're using self-hosted runners. But if that's not true, then my understanding about the cause was not correct. 
I tried to repro it by running different scenarios, like trying to run multiple jobs with the action but it worked for me every time. Let me know if you find out any more information about how this branch differs from the working branches. (May be try running it on other non-default branches?)

**Comment by scottwestover** [comment]
Thanks for the quick response. We haven't been able to pin down how this one branch/repo is different, will try to investigate further.

I also tried to recreate the issue, and have had no success in doing so. We are able to run this action on the `non-default` branch in this repo, and it downloads trivy without any issues. However, as soon as I setup this action on our `develop` branch, it fails. I also tried creating a new branch that is not the default branch, and I ran into the same issue.
